src/drivers/bmi055/bmi055_main.cpp

```diff
@@ -0,0 +1,527 @@
+
+
+    /* reset to default polling */
+    if (ioctl(fd_acc, SENSORIOCSPOLLRATE, SENSOR_POLLRATE_DEFAULT) < 0) {
+        err(1, "accel reset to default polling");
+    }
+
+    close(fd_acc);
+
+    /* reset to manual polling */
+    if (ioctl(fd_gyr, SENSORIOCSPOLLRATE, SENSOR_POLLRATE_MANUAL) < 0) {
+        err(1, "gyro reset to manual polling");
+    }
+
+
+    /* do a simple demand read */
+    sz = read(fd_gyr, &g_report, sizeof(g_report));
+
+    if (sz != sizeof(g_report)) {
+        warnx("ret: %d, expected: %d", sz, sizeof(g_report));
+        err(1, "immediate gyro read failed");
+    }
+
+    warnx("gyr x: \t% 9.5f\trad/s", (double)g_report.x);
+    warnx("gyr y: \t% 9.5f\trad/s", (double)g_report.y);
+    warnx("gyr z: \t% 9.5f\trad/s", (double)g_report.z);
+    warnx("gyr x: \t%d\traw", (int)g_report.x_raw);
+    warnx("gyr y: \t%d\traw", (int)g_report.y_raw);
+    warnx("gyr z: \t%d\traw", (int)g_report.z_raw);
+    warnx("gyr range: %8.4f rad/s (%d deg/s)", (double)g_report.range_rad_s,
+          (int)((g_report.range_rad_s / M_PI_F) * 180.0f + 0.5f));
+
+
+    /* reset to default polling */
+    if (ioctl(fd_gyr, SENSORIOCSPOLLRATE, SENSOR_POLLRATE_DEFAULT) < 0) {
+        err(1, "gyro reset to default polling");
+    }
+
+    close(fd_gyr);
+
+    /* XXX add poll-rate tests here too */
+    reset(external_bus);
+    errx(0, "PASS");
+
+}
+
+/**
+ * Reset the driver.
+ */
+void
+reset(bool external_bus)
+{
+    const char *path_accel = external_bus ? BMI055_DEVICE_PATH_ACCEL_EXT : BMI055_DEVICE_PATH_ACCEL;
+    const char *path_gyro = external_bus ? BMI055_DEVICE_PATH_GYRO_EXT : BMI055_DEVICE_PATH_GYRO;
+
+    int fd_acc = open(path_accel, O_RDONLY);
+
+    if (fd_acc < 0) {
+        err(1, "Opening accel file failed ");
+    }
+
+    int fd_gyr = open(path_gyro, O_RDONLY);
+
+    if (fd_gyr < 0) {
+        err(1, "Opening gyro file failed ");
+    }
+
+    if (ioctl(fd_acc, SENSORIOCRESET, 0) < 0) {
+        err(1, "accel driver reset failed");
+    }
+
+
+    if (ioctl(fd_acc, SENSORIOCSPOLLRATE, SENSOR_POLLRATE_DEFAULT) < 0) {
+        err(1, "accel driver poll restart failed");
+    }
+
+    close(fd_acc);
+
+    if (ioctl(fd_gyr, SENSORIOCRESET, 0) < 0) {
+        err(1, "gyro driver reset failed");
+    }
+
+    if (ioctl(fd_gyr, SENSORIOCSPOLLRATE, SENSOR_POLLRATE_DEFAULT) < 0) {
+        err(1, "gyro driver poll restart failed");
+    }
+
+    close(fd_gyr);
+
+    exit(0);
+}
+
+
+/**
+ * Print a little info about the driver.
+ */
+void
+info(bool external_bus)
+{
+    BMI055_accel **g_dev_acc_ptr = external_bus ? &g_acc_dev_ext : &g_acc_dev_int;
+    BMI055_gyro **g_dev_gyr_ptr = external_bus ? &g_gyr_dev_ext : &g_gyr_dev_int;
+
+    if (*g_dev_acc_ptr == nullptr) {
+        errx(1, "accel driver not running");
+    }
+
+    if (*g_dev_gyr_ptr == nullptr) {
+        errx(1, "gyro driver not running");
+    }
+
+    printf("state @ %p\n", *g_dev_acc_ptr);

```

Comment:
We could still print this if only the accelerometer is running and the gyro not.